### PR TITLE
search is dotcom: CTA fixes

### LINKED ISSUE #23923: CTA messaging for public code
CTAs need updating across the product including:
- [ ] Home page CTA (new design)
- [ ] Search results CTA (focus on searching public code, open in editor)
- [ ] Signup page copy edits (needs to reference public code)

Figma: https://www.figma.com/file/DzJsJq8DTBHUrauallnSwG/Public-code-CTA-updates?node-id=0%3A1&viewport=365%2C48%2C0.13

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #23923

### CODE REVIEW

**Review by limitedmage — APPROVED** [review]
Looks great, one more content change is required.
I also added the final blog post link as a blocker in #23866.

**Comment by limitedmage** [comment]
Let's hold off on merging this until we get confirmation from Marketing.

File: client/web/src/auth/ExperimentalSignUpPage.tsx [hunk 1]
```diff
@@ -22,12 +22,13 @@ interface Props extends ThemeProps, TelemetryProps {
...
 
 const SourceToTitleMap = {
-    Context: 'Easily search the code you care about.',
+    Context: 'Easily search all the code you care about.',
     Saved: 'Create a library of useful searches.',
     Monitor: 'Monitor code for changes.',
     Extend: 'Augment code and workflows via extensions.',
-    SearchCTA: 'Add your public (and soon private) repositories.',
-    Snippet: 'Easily search the code you care about.',
+    SearchCTA: 'Add your public and private repositories.',
```
Comment (rrhyne) [inline client/web/src/auth/ExperimentalSignUpPage.tsx:29]:
```suggestion
    SearchCTA: 'Easily search the code you care about.',
```

I think this is the correct copy for this var as per this design:
https://www.figma.com/file/DzJsJq8DTBHUrauallnSwG/Public-code-CTA-updates?node-id=4%3A6168

File: client/web/src/auth/ExperimentalSignUpPage.tsx [hunk 2]
```diff
@@ -22,12 +22,13 @@ interface Props extends ThemeProps, TelemetryProps {
...
+    HomepageCTA: 'Add your public and private repositories.',
```
Comment (rrhyne) [inline client/web/src/auth/ExperimentalSignUpPage.tsx:30]:
```suggestion
    HomepageCTA: 'Easily search the code you care about.',
```

**Review by rrhyne — APPROVED** [review]
I made two comments as to a sub-headline, but other than that everything looks great.

File: client/web/src/auth/ExperimentalSignUpPage.tsx (hunk 2 above)
Comment (limitedmage) [reply client/web/src/auth/ExperimentalSignUpPage.tsx]: @rrhyne  is it not "Easily search all the code you care about." (with "all")? Does the contexts one also need to change to remove "all"?

Comment (rrhyne) [reply client/web/src/auth/ExperimentalSignUpPage.tsx]: The suggestion is the final copy.

File: client/web/src/auth/ExperimentalSignUpPage.tsx [hunk 3]
```diff
@@ -22,12 +22,13 @@ interface Props extends ThemeProps, TelemetryProps {
 }
 
 const SourceToTitleMap = {
-    Context: 'Easily search the code you care about.',
+    Context: 'Easily search all the code you care about.',
```
Comment (rrhyne) [inline client/web/src/auth/ExperimentalSignUpPage.tsx:25]: Easily search the code you care about.

File: client/web/src/auth/ExperimentalSignUpPage.tsx [hunk 4]
```diff
@@ -22,12 +22,13 @@ interface Props extends ThemeProps, TelemetryProps {
...
     Saved: 'Create a library of useful searches.',
     Monitor: 'Monitor code for changes.',
     Extend: 'Augment code and workflows via extensions.',
-    SearchCTA: 'Add your public (and soon private) repositories.',
-    Snippet: 'Easily search the code you care about.',
+    SearchCTA: 'Easily search the code you care about.',
+    HomepageCTA: 'Easily search the code you care about.',
+    Snippet: 'Easily search all the code you care about.',
```
Comment (rrhyne) [inline client/web/src/auth/ExperimentalSignUpPage.tsx:31]:
```suggestion
    Snippet: 'Easily search the code you care about.',
```

File: client/web/src/auth/ExperimentalSignUpPage.tsx (hunk 2 above)
Comment (rrhyne) [reply client/web/src/auth/ExperimentalSignUpPage.tsx]: I think I got suggestions in on all the places all was used.

**Pull request merged**